File: backend/src/main/java/nlu/fit/backend/service/WishlistService.java
package nlu.fit.backend.service;

import nlu.fit.backend.dto.wishlist.MoveToCartRequest;
import nlu.fit.backend.dto.wishlist.WishlistItemResponseDto;

import java.util.List;

public interface WishlistService {
    List<WishlistItemResponseDto> getMyWishlist(String email);
    
    void addToWishlist(String email, Long productId);
    
    void removeFromWishlist(Long wishlistId, String email);
    
    void moveToCart(String email, MoveToCartRequest request);
}

File: frontend/src/modules/wishlist/service/wishlistService.ts
import { apiCall } from '../../../api/client';

export const wishlistService = {
  async getMyWishlist() {
    return await apiCall<any[]>('/wishlist', {
      method: 'GET',
      raw: true
    });
  },

  async addToWishlist(productId: number | string) {
    return await apiCall<any>(`/wishlist/${productId}`, {
      method: 'POST',
      raw: true
    });
  },

  async removeFromWishlist(wishlistId: number | string) {
    return await apiCall<any>(`/wishlist/${wishlistId}`, {
      method: 'DELETE',
      raw: true
    });
  },

  async moveToCart(wishlistIds: (number | string)[]) {
    return await apiCall<any>('/wishlist/move-to-cart', {
      method: 'POST',
      body: JSON.stringify({ wishlistIds }),
      raw: true
    });
  }
};
